packages/library-base/src/commonMain/kotlin/io/realm/query/RealmQuery.kt

```diff
@@ -0,0 +1,327 @@
+     * @param arguments Realm values for the predicate.
+     */
+    fun query(filter: String, vararg arguments: Any?): RealmQuery<T>
+
+    /**
+     * Sorts the query result by the specific property name according to [sortOrder], which is
+     * [Sort.ASCENDING] by default.
+     *
+     * Sorting is currently limited to character sets in 'Latin Basic', 'Latin Supplement',
+     * 'Latin Extended A', and 'Latin Extended B' (UTF-8 range 0-591). For other character sets
+     * sorting will have no effect.
+     *
+     * @param property the property name to sort by.
+     * @throws [IllegalArgumentException] if the property name does not exist.
+     */
+    fun sort(property: String, sortOrder: Sort = Sort.ASCENDING): RealmQuery<T>
+
+    /**
+     * Sorts the query result by the specific property name according to [Pair]s of properties and
+     * sorting order.
+     *
+     * Sorting is currently limited to character sets in 'Latin Basic', 'Latin Supplement',
+     * 'Latin Extended A', and 'Latin Extended B' (UTF-8 range 0-591). For other character sets
+     * sorting will have no effect.
+     *
+     * @param propertyAndSortOrder pair containing the property and the order to sort by.
+     * @throws [IllegalArgumentException] if the property name does not exist.
+     */
+    fun sort(
+        propertyAndSortOrder: Pair<String, Sort>,
+        vararg additionalPropertiesAndOrders: Pair<String, Sort>
+    ): RealmQuery<T>
+
+    /**
+     * Selects a distinct set of objects of a specific class. When multiple distinct fields are
+     * given, all unique combinations of values in the fields will be returned. In case of multiple
+     * matches, it is undefined which object is returned. Unless the result is sorted, then the
+     * first object will be returned.
+     *
+     * @param property first field name to use when finding distinct objects.
+     * @param extraProperties remaining property names when determining all unique combinations of
+     * field values.
+     * @throws IllegalArgumentException if [property] does not exist, is an unsupported type, or
+     * points to a linked field.
+     */
+    // TODO "points to a linked field" doesn't apply yet
+    fun distinct(property: String, vararg extraProperties: String): RealmQuery<T>
+
+    /**
+     * Limits the number of objects returned in case the query matched more objects.
+     *
+     * Note that when using this method in combination with [sort] and [distinct] they will be
+     * executed in the order they where added which can affect the end result.
+     *
+     * @param limit a limit that is greater than or equal to 1.
+     * @throws IllegalArgumentException if the provided [limit] is less than 1.
+     */
+    fun limit(limit: Int): RealmQuery<T>
+
+    /**
+     * Returns a query that finds the first object that fulfills the query conditions.
+     */
+    fun first(): RealmSingleQuery<T>
+
+    /**
+     * Finds the minimum value of a property.
```

Comment:
We should mention the reified variant in all of these similar to what you did for `Realm.query`